### ISSUE TITLE
Add Parameter type

### ISSUE THREAD

**Issue by tspooner** [tspooner/spaces#20]
The idea here is make use of `spaces` for parameter verification. For example, stochastic gradient descent (as used by many RL algorithms) uses a step size _alpha_ which should really be a value in the interval (0, 1]. At the moment, in all my code I do these checks manually but there may be a nicer way to automate this process using `spaces`.

It would be very nice to use the standard Rust builder pattern to construct parameter spaces as it were!

Note: this may well be best suited in a new crate that derives from `spaces`.

**Comment by tspooner** [comment]
This could actually replace the `Parameter` type I currently use in the `rsrl` framework.

**Comment by tspooner** [comment]
This will be dropped in favour of a more fully-featured implementation of parameter spaces. This will be a separate module and will likely include hyperparameter optimisation code.
